api/server/router/debug/debug.go

```diff
@@ -10,27 +10,23 @@ import (
 	"github.com/docker/docker/api/server/router"
 )
 
-// NewRouter creates a new debug router
-// The debug router holds endpoints for debug the daemon, such as those for pprof.
-func NewRouter() router.Router {
-	r := &debugRouter{}
-	r.initRoutes()
-	return r
-}
-
 type debugRouter struct {
 	routes []router.Route
 }
 
-func (r *debugRouter) initRoutes() {
-	r.routes = []router.Route{
-		router.NewGetRoute("/vars", frameworkAdaptHandler(expvar.Handler())),
```

Comment:
Actually; I've come to realise this is aligning with other routers; for those it's needed because the routes refer  to the router itself, for example: https://github.com/moby/moby/blob/9da4ae2ef3f8121a6ef0be213ad8f0d98bd63b91/api/server/router/volume/volume.go#L27-L31



For `debugRouter` that's  not currently  needed, but let's assume at some point it may, so we may as well keep the same pattern for now,  even if it's not yet needed.